cwl/src/main/scala/cwl/CommandLineTool.scala

```diff
@@ -1,66 +1,191 @@
-                                      String :+:
-                                      Array[
-                                        CwlType :+:
                                         CommandInputRecordSchema :+:
                                         CommandInputEnumSchema :+:
                                         CommandInputArraySchema :+:
                                         String :+:
-                                        CNil] :+:
-                                      CNil,
-                                    `type`: W.`"array"`.T,
-                                    label: Option[String],
-                                    inputBinding: Option[CommandLineBinding])
+                                        Array[
+                                          CwlType :+:
+                                            CommandInputRecordSchema :+:
+                                            CommandInputEnumSchema :+:
+                                            CommandInputArraySchema :+:
+                                            String :+:
+                                            CNil] :+:
+                                        CNil,
+                                      `type`: W.`"array"`.T,
+                                      label: Option[String],
+                                      inputBinding: Option[CommandLineBinding])
 
 
-case class CommandOutputParameter(
-                                   id: String,
-                                   label: Option[String] = None,
-                                   secondaryFiles: Option[Expression :+: String :+: Array[Expression :+: String :+: CNil] :+: CNil] = None,
-                                   format: Option[Expression :+: Array[String] :+: String :+: CNil] = None, //only valid when type: File
-                                   streamable: Option[Boolean] = None, //only valid when type: File
-                                   doc: Option[String :+: Array[String] :+: CNil] = None,
-                                   outputBinding: Option[CommandOutputBinding] = None,
-                                   `type`: Option[MyriadOutputType] = None)
+  case class CommandOutputParameter(
+                                     id: String,
+                                     label: Option[String] = None,
+                                     secondaryFiles: Option[Expression :+: String :+: Array[Expression :+: String :+: CNil] :+: CNil] = None,
+                                     format: Option[Expression :+: Array[String] :+: String :+: CNil] = None, //only valid when type: File
+                                     streamable: Option[Boolean] = None, //only valid when type: File
+                                     doc: Option[String :+: Array[String] :+: CNil] = None,
+                                     outputBinding: Option[CommandOutputBinding] = None,
+                                     `type`: Option[MyriadOutputType] = None)
```

Comment:
ToL and this is obviously preexisting, but it seems like there could be separate types here where these "only valid when type: File" attributes are only valid when the type is File

Reply:
As above.